### DISCUSSION TITLE
Use backpack/permissionmanager on V5

### DISCUSSION THREAD

**Discussion by Mohsin-Uforia** [Laravel-Backpack/community-forum#658] [Q&A (Help)]
I'm using Laravel v8.83.27 and can't update to v10, so I add backpack v5 instend of v6 that support Laravel 8. Currently, I'm attempting to incorporate 'backpack/permissionmanager' into my project by following the instructions: [](https://github.com/Laravel-Backpack/PermissionManager) . Unfortunately, I'm encountering an error

  Problem 1
    - Root composer.json requires backpack/permissionmanager ^7.0 -> satisfiable by backpack/permissionmanager[7.0.0, 7.0.1].
    - backpack/permissionmanager[7.0.0, ..., 7.0.1] require backpack/crud ^6.0 -> found backpack/crud[6.0.0, ..., v6.x-dev] but it conflicts with your root composer.json require (5.*).

I'm seeking guidance on how to successfully install 'backpack/permissionmanager' while using 'backpack-v5'.

I appreciate any assistance you can provide. Thank you in advance.

**Comment by karandatwani92** [comment]
Did you try to remove and re-install? to auto-pick the compatible version?
1) `composer remove backpack/permissionmanager`
2) `composer require backpack/permissionmanager`

**Comment by Mohsin-Uforia** [comment]
Thanks for reply 
I tryed to uninstall and then reinstall, but I am still facing the same error.
Here is commands out put


C:\xampp\htdocs\wishlist>composer remove backpack/permissionmanager
backpack/permissionmanager is not required in your composer.json and has not been removed
./composer.json has been updated
Running composer update backpack/permissionmanager
Loading composer repositories with package information
Package "backpack/permissionmanager" listed for update is not locked.
Updating dependencies
Nothing to modify in lock file
Writing lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package fruitcake/laravel-cors is abandoned, you should avoid using it. No replacement was suggested.
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Package fzaninotto/faker is abandoned, you should avoid using it. No replacement was suggested.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: backpack/crud
Discovered Package: creativeorange/gravatar
Discovered Package: digitallyhappy/assets
Discovered Package: facade/ignition
Discovered Package: fideloper/proxy
Discovered Package: fruitcake/laravel-cors
Discovered Package: laravel/tinker
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Discovered Package: prologue/alerts
Package manifest generated successfully.
80 packages you are using are looking for funding.
Use the `composer fund` command to find out more!



C:\xampp\htdocs\wishlist>composer require backpack/permissionmanager
Info from https://repo.packagist.org: #StandWithUkraine
Using version ^7.0 for backpack/permissionmanager
./composer.json has been updated
Running composer update backpack/permissionmanager
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires backpack/permissionmanager ^7.0 -> satisfiable by backpack/permissionmanager[7.0.0, 7.0.1].
    - backpack/permissionmanager[7.0.0, ..., 7.0.1] require backpack/crud ^6.0 -> found backpack/crud[6.0.0, ..., v6.x-dev] but it conflicts with your root composer.json require (5.*).

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require backpack/permissionmanager:*" to figure out if any version is installable, or "co
mposer require backpack/permissionmanager:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by karandatwani92** [comment]
Can you share your composer.json? I give it a try to install

**Comment by Mohsin-Uforia** [comment]
Sure, below is composer.json file code
Composer.json
{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The Laravel Framework.",
    "keywords": [
        "framework",
        "laravel"
    ],
    "license": "MIT",
    "require": {
        "php": "^7.3",
        "fideloper/proxy": "^4.2",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "laravel/framework": "^8.0",
        "laravel/tinker": "^2.0"
    },
    "require-dev": {
        "facade/ignition": "^2.3.6",
        "fzaninotto/faker": "^1.9.1",
        "mockery/mockery": "^1.3.1",
        "nunomaduro/collision": "^5.0",
        "phpunit/phpunit": "^9.3"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ]
    }
}
